### PR TITLE
refactor: event advanced web wrapper

### PR DESCRIPTION
**Pull request by ibex088**
## What does this PR do?

- Introduced a new EventAdvancedWebWrapper component to centralize all TRPC endpoint calls, improving data handling and separation of concerns.
- Refactored the existing EventAdvancedTab and all its children components into (dumb) components, shifting all TRPC calls and logic to the newly created parent wrapper (EventAvailabilityTabWebWrapper). It now solely handles UI rendering, receiving data and callbacks as props from its parent.

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export type EventAdvancedWebWrapperProps = Pick<EventTypeSetupProps, "eventType" | "team">;
+
+const EventAdvancedWebWrapper = (props: EventAdvancedWebWrapperProps) => {
+  const connectedCalendarsQuery = trpc.viewer.connectedCalendars.useQuery();
+  const { data: user, isPending } = trpc.viewer.me.useQuery();
+  return (
+    <EventAdvancedTab
+      {...props}
+      connectedCalendars={connectedCalendarsQuery.data}
```
Comment (ThyMinimalDev) [inline packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx:13]: let's rename this props to calendarsQueryData

File: apps/web/modules/settings/my-account/appearance-view.tsx [hunk 2]
```diff
@@ -304,6 +304,7 @@ const AppearanceView = ({
               description={t("bookerlayout_user_settings_description")}
               isDisabled={isBookerLayoutFormSubmitting || !isBookerLayoutFormDirty}
               isLoading={mutation.isPending}
+              user={user}
```
Comment (ThyMinimalDev) [inline apps/web/modules/settings/my-account/appearance-view.tsx:307]: can you give some context on this change ?

File: packages/features/settings/BookerLayoutSelector.tsx [hunk 3]
```diff
@@ -108,9 +116,11 @@ const BookerLayoutFields = ({
   showUserSettings,
   isDark,
   isOuterBorder,
+  user,
+  isUserLoading,
 }: BookerLayoutFieldsProps) => {
   const { t } = useLocale();
-  const { isPending: isUserLoading, data: user } = useMeQuery();
+  // const { isPending: isUserLoading, data: user } = useMeQuery();
```
Comment (SomayChauhan) [inline packages/features/settings/BookerLayoutSelector.tsx:123]: removed this query. now sending from the parent as prop.

File: apps/web/modules/settings/my-account/appearance-view.tsx (hunk 2 above)
Comment (SomayChauhan) [reply apps/web/modules/settings/my-account/appearance-view.tsx:307]: https://github.com/calcom/cal.com/pull/16711#discussion_r1770839258

File: packages/features/settings/BookerLayoutSelector.tsx (hunk 3 above)
Comment (SomayChauhan) [reply packages/features/settings/BookerLayoutSelector.tsx:123]: https://github.com/calcom/cal.com/pull/16711/commits/f28430e29737e02eb9c144e4910b257c69c750ef

File: packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx (hunk 1 above)
Comment (SomayChauhan) [reply packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx]: https://github.com/calcom/cal.com/pull/16711/commits/f28430e29737e02eb9c144e4910b257c69c750ef

File: packages/features/settings/BookerLayoutSelector.tsx (hunk 3 above)
Comment (ThyMinimalDev) [reply packages/features/settings/BookerLayoutSelector.tsx:123]: let's remove the comment if we dont need this anymore

File: packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { EventAdvancedTab } from "@calcom/features/eventtypes/components/tabs/advanced/EventAdvancedTab";
+import type { EventTypeSetupProps } from "@calcom/features/eventtypes/lib/types";
+import { trpc } from "@calcom/trpc/react";
+
+export type EventAdvancedWebWrapperProps = Pick<EventTypeSetupProps, "eventType" | "team">;
+
+const EventAdvancedWebWrapper = (props: EventAdvancedWebWrapperProps) => {
+  const connectedCalendarsQuery = trpc.viewer.connectedCalendars.useQuery();
+  const { data: user, isPending } = trpc.viewer.me.useQuery();
```
Comment (SomayChauhan) [inline packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx:9]: will move further up the tree after https://github.com/calcom/cal.com/pull/16808#discussion_r1780564577 is merged

File: packages/features/settings/BookerLayoutSelector.tsx (hunk 3 above)
Comment (SomayChauhan) [reply packages/features/settings/BookerLayoutSelector.tsx]: https://github.com/calcom/cal.com/commit/7300c9de125390dc7a87d7a9162cc714508097bf

File: packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx (hunk 4 above)
Comment (SomayChauhan) [reply packages/platform/atoms/event-types/wrappers/EventAdvancedWebWrapper.tsx]: https://github.com/calcom/cal.com/commit/e18da6c4a51ad3b7028dc6a4e5e95d94d9a6deff

**Review by ThyMinimalDev — APPROVED** [review]

**Pull request merged**